NetCoreTeamCity/Services/TestOccurrencesService.cs

```diff
@@ -0,0 +1,76 @@
+﻿using NetCoreTeamCity.Api;
+using NetCoreTeamCity.Clients;
+using NetCoreTeamCity.Exceptions;
+using NetCoreTeamCity.Locators;
+using NetCoreTeamCity.Locators.Build;
+using NetCoreTeamCity.Locators.TestOccurrences;
+using NetCoreTeamCity.Models;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+
+namespace NetCoreTeamCity.Services
+{
+    internal class TestOccurrencesService
+    {
+        private readonly ITeamCityApiClient ApiClient;
+        private string endpoint = "testOccurrences";
+
+        public TestOccurrencesService(ITeamCityApiClient apiClient)
+        {
+            ApiClient = apiClient;
+        }
+
+        public TestOccurrence Get(string id)
+        {
+            try
+            {
+                return ApiClient.Get<TestOccurrence>($"{endpoint}/{id}");
+            }
+            catch (HttpException exception)
+            {
+                if (exception.StatusCode == HttpStatusCode.NotFound) return null;
+                throw;
+            }
+        }
+
+        public IList<TestOccurrence> Find(BuildLocator locator)
+        {
+            string query = GetQuery(By.TestOccurences.Build(locator));
+
+            var testOccurreces = ApiClient.Get<TestOccurrences>($"{endpoint}{query}");
+            return testOccurreces.TestOccurrenceItems == null ? new List<TestOccurrence>() : testOccurreces.TestOccurrenceItems;
+        }
+
+        public IList<TestOccurrence> Find(TestOccurrencesLocator locator)
```

Comment:
This method wont be needed, please leave only methods that would use TestOccurrenceLocator and BuildLocator + (with optional fields param)